### ISSUE TITLE
Rugged::SshError (error: git-shell died of signal 13

### ISSUE THREAD

**Issue by bougui** [jbox-web/redmine_git_hosting#700]
Hi there,

we are on centos 7, redmine 3.4.3.stable, gitolite3 v3.6.7-15-g09faacb redmine_git_hosting-1.2.3 see all the version bellow.

This error is seen on an on a empty redmine installation, 

All went fine for the plugin installation except that there is a gem requirement conflit between 

redmine-3.4.3 which require redcarpet 3.4.0

And this plugin which ask for redcarpet 3.3.2

To resolve this we have change the plugin Gemfile requirement on line 7 to

> gem 'redcarpet', '~> 3.4.0'

Is this fine, or are we going after trouble here ?

After that, all seem fine, we can go into the plugin configuration tabs and put the correct hostname and the correct ssh key path.

We can install the hook and in the status page all is green.

But then we we try to create a gitolite repo we always get Internal error . All other task in redmine works fine.   But all git_hosting action now end in internal error 500 `(error: git-shell died of signal 13`

Do we need a special sshd_config setup on Centos 7 ?

Here are the error from production.log file.

```
Completed 500 Internal Server Error in 568ms (ActiveRecord: 3.9ms)

Rugged::SshError (error: git-shell died of signal 13
s/master
0000):
  plugins/redmine_git_hosting/lib/redmine_git_hosting/gitolite_wrappers/global/enable_rw_access.rb:16:in `call'
  plugins/redmine_git_hosting/lib/redmine_git_hosting/gitolite_wrappers/base.rb:24:in `call'
  plugins/redmine_git_hosting/lib/redmine_git_hosting/gitolite_wrapper.rb:83:in `call_gitolite_wrapper'
  plugins/redmine_git_hosting/lib/redmine_git_hosting/gitolite_wrapper.rb:60:in `execute_action'
  plugins/redmine_git_hosting/lib/redmine_git_hosting/gitolite_wrapper.rb:24:in `resync_gitolite'
  plugins/redmine_git_hosting/lib/redmine_git_hosting/gitolite_accessor.rb:146:in `resync_gitolite'
  plugins/redmine_git_hosting/lib/redmine_git_hosting/gitolite_accessor.rb:117:in `enable_rw_access'
  plugins/redmine_git_hosting/app/use_cases/settings/apply.rb:170:in `do_add_redmine_rw_access'
  plugins/redmine_git_hosting/app/use_cases/settings/apply.rb:56:in `apply_settings'
  plugins/redmine_git_hosting/app/use_cases/settings/apply.rb:36:in `call'
  plugins/redmine_git_hosting/app/use_cases/settings/apply.rb:28:in `call'
  plugins/redmine_git_hosting/lib/redmine_git_hosting/patches/settings_controller_patch.rb:79:in `execute_post_actions'
  plugins/redmine_git_hosting/lib/redmine_git_hosting/patches/settings_controller_patch.rb:69:in `handle_settings_update'
  plugins/redmine_git_hosting/lib/redmine_git_hosting/patches/settings_controller_patch.rb:41:in `plugin_with_git_hosting'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'
```

And in the git_hosting.log all seem fine.

This is the syslog entry generated by gitolite
```
Oct 27 08:45:19 lpsredmine102 gitolite[28642][28642]: ssh#011ARGV=redmine_gitolite_admin_id_rsa#011SOC=git-receive-pack '/gitolite-admin.git'#011FROM=127.0.0.1
Oct 27 08:45:19 lpsredmine102 gitolite[28642][28642]: pre_git#011gitolite-admin#011redmine_gitolite_admin_id_rsa#011W#011any#011refs/.*
Oct 27 08:45:19 lpsredmine102 gitolite[28642][28658]: update#011gitolite-admin#011redmine_gitolite_admin_id_rsa#011W#011refs/heads/master#0114a60badb4a41815be43442045577375e34559f33#011e0e014c58de0017ad6e6c150b66f6af804864d28#011refs/.*
```

This is our test environnement:
```
Environment:
  Redmine version                3.4.3.stable
  Ruby version                   2.4.2-p198 (2017-09-14) [x86_64-linux]
  Rails version                  4.2.8
  Environment                    production
  Database adapter               Mysql2
SCM:
  Subversion                     1.7.14
  Git                            1.8.3.1
  Filesystem                     
  Xitolite                       1.8.3.1
Redmine plugins:
  additionals                    2.0.4
  clipboard_image_paste          1.12
  redmine_activerecord_session_store 0.0.1
  redmine_bootstrap_kit          0.2.5
  redmine_drawio                 0.8.0
  redmine_git_hosting            1.2.3
  redmine_lightbox2              0.4.3
  redmine_silencer               0.4.2
  redmine_wiki_extensions        0.8.1
  sidebar_hide                   0.0.8
```

Any clue would be appreciate.

Thanks

**Comment by stevetai** [comment]
I have almost the same environment: CentOS 7, Redmine 3.4.3.stable.17120, Ruby 2.4.1, MySql2, Git 1.8.3.1 (Also manually change redcarpet to 3.4.0), and the same issue occurs to me.
After spending several days to trace, try and error, finally I find a way to fix it.

I found that, as logging in to redmine account (su redmine) and change something manually under /tmp/redmine_git_hosting/git/gitolite-admin.git/ and git push will be fail (error: git-shell died of signal 13, fatal: error in sideband demultiplexer). So I reinstall gitolite and add my own ssh key as initial admin key, and try to modify gitolite-admin again. but fail again.

Turn out I found that the latest Git version is 2.15.1, which is quite different from mine (1.8.3.1) which installed through yum. Since mine is the latest one in yum, I just uninstall Git through 'yum remove' and install 2.15.1 manually (by following this [instruction](https://blacksaildivision.com/git-latest-version-centos) and relogin the console to check git version to make sure if it's updated.). There is no need to reinstall gitolite. 

First I try to use git and gitolite on console to make sure git pull, push command works, then login to Redmine. Then issue #691 occurs, i.e.: in Settings, it shows error for Xitolite. So I followed this [modification #697](https://github.com/jbox-web/redmine_git_hosting/pull/697/files). Then everything is perfect.

Hope these can help you.
